extension/oidcauthextension/extension.go

```diff
@@ -139,8 +159,25 @@ func (e *oidcExtension) Authenticate(ctx context.Context, headers map[string][]s
 	return client.NewContext(ctx, cl), nil
 }
 
-func (e *oidcExtension) setProviderConfig(ctx context.Context, config *Config) error {
-	e.transport = &http.Transport{
+func (e *oidcExtension) resolveProvider(issuer string) (*ProviderContainer, error) {
+	if len(e.providerContainers) == 1 {
+		for _, providerContainer := range e.providerContainers {
+			return providerContainer, nil
+		}
```

Comment:
Since we have already confirmed that there's a provider available
```suggestion
		return e.providerContainers[0], nil
```

Reply:
🤦

Reply:
Ah, of course, makes sense. Could do maps.Values(...)[0] but I don't think that really nets us much. 👍